app/models/claims/training_allowance.rb

```diff
@@ -0,0 +1,54 @@
+class Claims::TrainingAllowance
+  def initialize(mentor:, provider:, academic_year:, claim_to_exclude: nil)
+    @mentor = mentor
+    @provider = provider
+    @academic_year = academic_year
+    @claim_to_exclude = claim_to_exclude
+  end
+
+  def training_type
+    has_initial_training? ? REFRESHER_TRAINING : INITIAL_TRAINING
+  end
+
+  def total_hours
+    training_type == INITIAL_TRAINING ? INITIAL_TRAINING_HOURS : REFRESHER_TRAINING_HOURS
+  end
+
+  def remaining_hours
+    total_hours - hours_completed
+  end
+
+  def available?
+    remaining_hours.positive?
+  end
+
+  private
+
+  attr_reader :mentor, :provider, :academic_year, :claim_to_exclude
+
+  INITIAL_TRAINING = :initial
+  INITIAL_TRAINING_HOURS = 20
+  REFRESHER_TRAINING = :refresher
+  REFRESHER_TRAINING_HOURS = 6
+
+  private_constant :INITIAL_TRAINING, :INITIAL_TRAINING_HOURS, :REFRESHER_TRAINING,
+                   :REFRESHER_TRAINING_HOURS
+
+  def mentor_training_scope
+    @mentor_training_scope ||= provider.mentor_trainings
+                                       .where(mentor:)
+                                       .joins(claim: { claim_window: :academic_year })
+  end
+
+  def has_initial_training?
+    mentor_training_scope.where(claims: { status: :submitted, claim_windows: { academic_years: { ends_on: ..academic_year.starts_on } } }).exists?
```

Comment:
We need to be careful about hardcoding `status: :submitted` here. Dan is currently in the process of adding new statuses which presumably need to be included in this query – for example, see #1027.

I wonder if we'd be better off excluding the statuses we know should be ignored:

```ruby
.where.not(claims: { status: %i[draft internal_draft] })
```

But then I'm not entirely confident that every other status should be considered as valid in this context.

@Nitemaeric can you advise which statuses to filter out here?

I also wonder if there's a meaningful grouping for these statuses – for example, could they be defined as a scope on the `Claim` model, or even an array defined as a constant? Or is this only relevant here in this particular class?

Reply:
Yeah, I'd typically urge towards explicitly listing the statuses, rather than excluding them.

But in this case, the list of statuses isn't complete.

For statuses used for calculations, anything that gets to the "submitted" status and beyond would all be valid.

So in this instance, the code block you've suggested works.